core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -70,15 +70,11 @@
 
   public static final Charset US_ASCII = Charset.forName("US-ASCII");
 
-  private static boolean isAppengineEnvironment() {
-    String appengineEnvironment = System.getProperty("com.google.appengine.runtime.environment");
-    return "Production".equals(appengineEnvironment) || "Development".equals(appengineEnvironment);
-  }
-
-  // Certain production AppEngine runtimes have constraints on threading and socket handling
+  // AppEngine runtimes have constraints on threading and socket handling
   // that need to be accommodated.
   public static final boolean IS_RESTRICTED_APPENGINE =
-      isAppengineEnvironment() && "1.7".equals(System.getProperty("java.specification.version"));
+      System.getProperty("com.google.appengine.runtime.environment") != null
```

Comment:
FWIW - I did talk to the AppEngine folks about Java 8 support and the expectation is that the runtime classloader is more lenient.

@lesv  who keeps an eye on this stuff